Activity new clients for current month docs (#16472)
hashicorp/vault: website/content/docs/concepts/client-count/index.mdx

For the UI to be able to modify the configuration settings, it additionally needs `update` permission to
`sys/internal/counters/config`.

## New Clients For Current Month

> As of Vault 1.12, the billing period for the activity log API can be specified to include the current month 

Author: Notes from Vishal sync: we should add a disclaimer/small paragraph explaining the reason for the approximation (ie why it's necessary) without getting too technical, but explaining that it's a purely technical reason, a general computing limitation.

Author: Make sure to stress that this is done primarily to improve API performance and make the UI usable. Mention the problem name (count-distinct for multisets) and explain why this maps to that.

For the UI to be able to modify the configuration settings, it additionally needs `update` permission to
`sys/internal/counters/config`.

## New Clients For Current Month

> As of Vault 1.12, the billing period for the activity log API can be specified to include the current month 

Reviewer: ```suggestion
The billing period for the activity log API can be specified to include the current month 
```

for the end date. For more information about how this can be done, please visit 
[the internal counters API docs](/api-docs/system/internal-counters.mdx). 

When the end date is the current month, the `new_clients` counts will be an approximation of the true 

> number of new clients for the month, not an exact value. Note that the `new_clients` counts for the rest 

Reviewer: ```suggestion
number of new clients for the month, and not an exact value. Note that the `new_clients` counts for the rest 
```

> of the months will all be accurate. 

Reviewer: ```suggestion
of the months will be accurate. 
```

### Why An Approximation?

The `new_clients` counts for the current month is an approximation primarily to improve API 

> performance and make the UI usable. In order to give an exact value for the current month's

Reviewer: ```suggestion
performance and make the UI usable. To give an exact value for the current month's
```

new clients, client IDs need to be de-duplicated with the previous months' client IDs, 
which is a time and i/o intensive process. 

### Approximation Details And Accuracy Testing Results

> The `new_clients` approximation is calculated is by using a [hyperloglog algorithm](https://en.wikipedia.org/wiki/HyperLogLog)

Reviewer: ```suggestion
The `new_clients` approximation is calculated by using a [hyperloglog algorithm](https://en.wikipedia.org/wiki/HyperLogLog)
```

> to approximate the cardinality of the total number of clients within the billing period, and the cardinality 

Reviewer: ```suggestion
to approximate the cardinality of the total number of clients within the billing period. The cardinality 
```

> of the total number of clients within the billing period not including the current month. The returned value 

Reviewer: ```suggestion
of the total number of clients within the billing period does not the current month. The returned value 
```

is the difference between these two numbers.

The hyperloglog library used for the cardinality estimate is [axiomhq](https://github.com/axiomhq/hyperloglog),

> with 14 registers and the use of sparse representations when applicable. Some accuracy estimates can be found 

Reviewer: ```suggestion
with fourteen registers and the use of sparse representations, when applicable. Some accurate estimates can be found 
```

in this library's [README](https://github.com/axiomhq/hyperloglog#readme). These are accuracy results
for the cardinality of the multiset, which is the total number of clients within a billing period. The accuracy 
estimate for the number of new clients can be far lower, depending on the discrepancy between the number of 
clients in the current month and the number of clients in the billing period. 

If we call the number of clients for the current month C and the number of clients in the billing period B, we 
have found that in general, if C << B, the approximation can be imprecise, and the further the difference between
C and B grows, the more imprecise the approximation will be. Furthermore, the closer C is to 0, the more imprecise

> the approximation can be. Also, the more months in the billing period the less accurate the approximation can be. 

Reviewer: ```suggestion
the approximation can be. Also, the more months in the billing period, the less accurate the approximation can be. 
```
